jme3-core/src/main/java/com/jme3/anim/AnimationMask.java

```diff
@@ -31,12 +31,14 @@
  */
 package com.jme3.anim;
 
+import com.jme3.export.Savable;
+
 /**
  * Created by Nehon
  * An AnimationMask is defining a subset of elements on which an animation will be applied.
  * Most used implementation is the ArmatureMask that defines a subset of joints in an Armature.
  */
-public interface AnimationMask {
+public interface AnimationMask extends Savable {
```

Comment:
This change breaks the legacy code. Do not modify the interface. Please remove the Savable extension from the AnimationMask interface and move it to the ArmatureMask class.

Reply:
Thanks for the comment. Hmm. Can you explain in what way? It's only used by ArmatureMask it seems, and only created during runtime (since it's not saved)

Reply:
The purpose of the AnimationMask interface is to enable anyone to create their own implementation (I’ve written my own as well). This change, however, makes previous code non-compilable. The ArmatureMask class is just one of many implementations, so don’t focus solely on that one. Serialization of AnimChannels wasn’t provided in the old animation system either, so the absence of AnimLayer/AnimationMask serialization is a deliberate design choice. While the graphics editor in the SDK can help with mixing animations on multiple layers, I don’t think it’s necessary to save it. Multi-level animation is rarely used and complicated to manage in games. Nonetheless, it’s a feature that anyone can implement using a JSON file and a Loader designed according to their needs.

Reply:
In my own case, I setup blending in my asset pipeline but it won't get stored... so I have to store a completely separate descriptor file to load it into my game so that I can blend animations.  It's a little frustrating... but I've already worked around it so I don't personally have a dog in this fight.

This is only tangentially related to anim layers... but it is frustrating that the only thing that gets saved/loaded is the stuff loaded directly from external asset files.  None of the JME "value add" gets stored.

Reply:
How about making ArmatureMask Savable then?

Even if it was a "deliberate design choice", is there a reason it can't be changed? Would it be a problem if this change occurred? Even if it did break backwards compatibility, how many people would be affected? 3? (Which, arguably, is still more than would use this feature)

`Multi-level animation is rarely used`
I'd say any modern fps or 3rd person shooter has multi-level animations. Not to mention IK based on those layers.

`and complicated to manage in games.`
Hence my desire to make tools to simplify them! Being able to view, edit and save them removes several steps a developer has to do.

`it’s a feature that anyone can implement`
Anyone can implement any feature, so why make any tools at all?

Reply:
I'm against changing the interface because that forces everyone to be savable even if they don't want to be but I'm a fan of changing the JME implementations to be savable.  Unless someone can come up with a good reason why this is a bad idea... JME should drink it's own kool-aid here.